### ISSUE TITLE
Adjustments to BASE's latest descriptions

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#193]
In preparation for next interviews (see also issue #188 )
This is issue so for edits or adjustments to descriptions in _Balloons and Static Electricity_, based on content in latest version (version 1.2.0-reader.7).

- [x] Ensure all mentions of charges are explicit, i.e. no use of "positive ones" or "negative ones".
- [x] Ensure there is a sentence or indication in initial Scene Summary (and reset Scene Summary) the position of the wall and the sweater.
- [x] Add a grabbed state description to front of balloon description upon grabbing (also mentioned in #183)
- [x] Change all instances of "spacebar" to "space bar" (I just noticed bad pronunciation of it as one word.) 
- [x] If not already done, please add a hyphen to Two-balloon Experiment, switch.

These are issues, I am recording from memory. I can add specifics, once I re-listen.

**Comment by jessegreenberg** [comment]
> Ensure all mentions of charges are explicit, i.e. no use of "positive ones" or "negative ones".

What should they be?  Ones -> charges? For instance, in the first cognitive walkthrough example 
> "Yellow Balloon. 
*In center of play area.
*Has net neutral charge, no more negative charges than positive ones.
*Press W, A, S, or D key to drag balloon.
*Spacebar, to let go. 
*H key for hot keys and help. "

Should positive ones be changed to positive charges?

**Comment by terracoda** [comment]
@jessegreenberg, yes, final descriptions are in PDOM tab. I haven't had a chance to update all the walkthroughs, sorry :-)  After discussing with Emily, we decided to not use "ones" at all, just always use "charges".

**Comment by jessegreenberg** [comment]
> Ensure there is a sentence or indication in initial Scene Summary (and reset Scene Summary) the position of the wall and the sweater

Could you provide an example of whta this content should be?  In row 9 of the scene summary descriptions outlined in  https://docs.google.com/spreadsheets/d/1BiXFN2dRWfsjqV2WvKAXnZFhsk0jCxbnT0gkZr_T5T0/edit?ts=568067c0#gid=0, it seems that location of sweater and wall are only meant to be described when a balloon is touching or near one of them.

**Comment by jessegreenberg** [comment]
Ahh, I see on row 6, page load is supposed to have a list item:

* Yellow balloon, in center of play area, evenly between sweater and wall. Sweater is at far left, wall at far right.

**Comment by terracoda** [comment]
You got it @jessegreenberg, that is the missing content :-)

**Comment by jessegreenberg** [comment]
Thanks, sorry!

**Comment by terracoda** [comment]
@jessegreenberg, there also might be a blank list item in the list. VO reads out an empty item.

**Comment by terracoda** [comment]
@jessegreenberg, and no need to be sorry :-)

**Comment by jessegreenberg** [comment]
Marking as high priority, I think this should be done for #188.

**Comment by terracoda** [comment]
@jessegreenberg, I am just going to put my notes here of a test here for now to see how that works:
Mac OS Sierra, VO and Safari 10.0.1
__Loading the sim and auto read through__
- VO shows the number of headings, form controls, and live regions, in the VO captions box, but doesn't voice it out- very strange, but likely not sim bug.
- As suggested by VO, I used the Control-Option-Shift and the down arrow to interact with web content. I only pressed the down arrow once and VO read out all the content except the Scene Summary. It read the heading in the Scene Summary (skipped the list), and then proceeded to read out all the content from the Play Area heading to the Reset All button.
- VO announced Yellow Balloon at the end of the Grab yellow balloon button description again before announcing the Wall. I assume this is because the balloon is an application (a region-like thing. VO did not say "region" which is nice).
- VO announced "article" after the reset button.

__Listening to content item by item__
- I use only the Control-Option-Right Arrow to read forward, or the Control-Option-Left Arrow to review (go back). Using these VO keys and the Right Arrow, I can get to the Scene Summary list.
- 5 items read out fine (no blank lists read out with test on Sierra).
- VO reads out Play Area and Sweater (and description fine)
- At the end of the Balloon description VO reads out "dimmed"
- A couple of times, VO repeated the balloon description twice before it went onto the Grab Yellow balloon button. This only happened twice. Repeat tries, the balloon description was only read out once (strange VO behaviour maybe?)
- VO skipped the Wall description once, but it was there when I press the Control-Option-Left Arrow
- VO also says "dimmed" at the end of the Wall description.
- VO reads out switch as "Two Balloon Experiment, off, switch" then after pressing the VO command Control-Option-Space "on, Two Balloon Experiment, switch", then again,  "off, Two Balloon Experiment, switch"
- VO doesn't show the hyphen in Two-balloon Experiment, maybe, it is not there yet. I'll add to check list.
- The Interaction Alert "Balloon added/removed" was not announced, but I did Control-Option-Right Arrow and found it in the PDOM. But then couldn't repeat this. Alerts do not work on my Sierra Machine (don't know why yet).

__Balloon Grab Focus__
- After grabbing balloon, I can use CO-Right or Left arrow to read elsewhere, after to arrowing back to the balloon the black Grab focus is still there, but I cannot use the WASD keys to move it. I need to hit Space bar of Enter Key again to activate Application again (interesting). (Maybe we can address this in a the help documentation)

__Tab Through__
- sounds good: Grab Balloon button

Now will test on my other Mac and in ChromeCox to hear the Interaction Alerts.

**Comment by terracoda** [comment]
__Summary of these Notes Above__
- Sim sounded pretty good. @jessegreenberg, I added the hyphen issue to the check list.
- I don't understand what VO means by "dimmed", but it seems to happen on a group of things. It might be interesting to investigate this (later).
- The grab focus is an issue, but maybe we can handle that by watching "Application" mode. Is there anyway to track that? Likely not since that is a screen reader thing.

Now onto listening to the Interaction Alerts.

**Comment by terracoda** [comment]
@jessegreenberg, now adding notes for my run through on my 2011 MackBook Pro, MacOS Yosemite (not Mavericks), VO and Safari.

**Comment by jessegreenberg** [comment]
Great, thanks @terracoda. 

> The grab focus is an issue, but maybe we can handle that by watching "Application" mode.

I can't think of any way to watch modes, screen readers don't have an API for this kind of communication in a browser.

**Comment by jessegreenberg** [comment]
It seems that 'dimmed' is VoiceOver's way of saying that something is disabled. http://maxdesign.com.au/jobs/sample-accessibility/05-forms/attribute-disabled.html.  Agreed that we should investigate this.

**Comment by terracoda** [comment]
@jessegreenberg 
Notes for my run through on 2011 MackBook Pro, MacOS Yosemite, VO and Safari.

__Page Load and Auto Read__
- [ ] VO started reading PDOM content without the CO-Arrow keys. It read all the Scene Summary content and everything until it got to the balloon. It got stuck there and kept repeating "Balloons and Static Electricity" several times until I hit Escape key. On page reload, VO did not get stuff and read everything.

__Read through with VO cursor keys (CO-L/R Arrow)__
- Do we have periods at the end of the descriptions? If not, I think we should add them. 
- Balloon descriptions is not always readable with cursor keys with this system. I got to it by reading backwards (CO-Left Arrow). This is strange. 
- And I heard text, blank when looking to Balloon description and once when looking for Wall description.
- With is read out as "Two Balloon Experiment". VO help text identifies it as a switch. Pressing with Space bar alone or the VO command CO-Space activates Green balloon and fires Interaction Alert.
- Listening through was good, except the balloon description was not found in a regular way.

__Controls not Read Out__
- Heading levels, Buttons, and switch are not read out on access, but the VO tooltip identifies them correctly after a few seconds (e.g., "You are currently on a switch, etc."). This might be due to my VO settings. I might have it on low - will check.

__Balloon Grab Focus__
- Upon grabbing, the balloon description is 

__Visual out of Synch__
This might be a performance issue on my computer? It's an interesting issue. I will try to reproduce on my other machine.
- I had the negative green balloon grabbed, and it was against the wall. I used the cursor keys to go over to the Remove Wall button. I removed the Wall. The balloon's release Interaction Alert fired, but the Green Balloon did not attract to the sweater. I used the Arrow keys to try to get back to the balloon, and tried a couple of keys (not sure what I did), and some how released the balloon and it attracted to sweater without an interaction alert.

__Rubbing Interaction Alerts__
- Rubbing alerts were good, but the last one (when sweater is depleted) was not read. VO read out the help tag, but seemed like the description of the opposite balloon. This might be specific to having two negatively charged balloons. Needs further testing, but not before next interviews.
- The depleted sweater works when there is a single balloon.

__Rubbing with Single Balloon__
- might need to fire full charge updates, we can test as is, though.

__Jump Key position Alerts__
- JS fires a position alert, JC fires the position alert, and JW fires with both At Wall and the Wall description
- [ ] JN position does not fire. This should announce "Near Wall"
- I think JN when wall is present should Announce "Near Wall." And eventually we will need induced charged description. Taliesin will have to create a description. It might be similar to one already in a walk through.

@jessegreenberg stopping for dinner, will continue and summarize in the morning :-)

**Comment by jessegreenberg** [comment]
For
>  Add a grabbed state description to front of balloon description upon grabbing (also mentioned in #183)

We are going to explicitly add 'Grabbed' to the description, so I am removing the aria-grabbed attribute.  It has poor support and is to be deprecated from the ARIA specification anyway.

**Comment by terracoda** [comment]
@jessegreenberg, I think this is a good idea 👍

**Comment by jessegreenberg** [comment]
Ok, good, thanks!

**Comment by jessegreenberg** [comment]
> stopping for dinner, will continue and summarize in the morning

@terracoda these are excellent reviews, thanks for doing this!  I think it would be great to go through this list and create new issues for the problems that you think are worth investigating.  The original checklist is complete, so I am going to remove the high priority label and assign to you.  If any other descriptions need to be changed, please let me know and reassign back to me.

**Comment by terracoda** [comment]
Awesome work @jessegreenberg!
Yes, still want to review them a little, but the high priority stuff is done.

**Comment by terracoda** [comment]
@jessegreenberg, I want to do a similar review for ChromeVox on both systems I have as well. I am assuming it's ok to do these reviews right here in the issue. I like the idea of the issue because it time stamps it. Maybe next time, I'll create a separate issue for testing notes and label it as such.

**Comment by terracoda** [comment]
@jessegreenberg
Notes for my run through on 2015 MackBook Air, MacOS Sierra, Chrome 54 with ChromeVox.

__Page Load and Auto Read__
- On first load, the content started reading automatically.
- On page reload, just the h1 was read out, then i need to use the CV-cursor keys to read line by line.

__Balloon Widget Interactions (focus)__
- On each CV-cursor down, I hear the content of each new element... 
- First the h3, then the description, then the button, then the button's help description, then I hear "application", then "Yellow Balloon." I heard this before in previous tests with VO, but I didn't realize what was causing it. We will see if this is better once @jessegreenberg implements the new structure mentioned in issue #187 
- Using CV-cursor down the pink focus highlights on grab button, off for help description, then on again on the empty application div. The grab interaction works perfectly while pink focus highlight is on the button. I did not expect it to work on the application div, but it sort of does partially. If I hit space bar twice, I grab the balloon. It seems to go to the button on the first press, then grab on the second press. Pressing the Enter key (while on the application div) any number of times does not grab the balloon. This leads me to believe that we may be able to make this work better. Again, we can compare how this works after the structure change mentioned in #187
- Note using CV-cursor forward and back onto the application div with the black grabbed state, I can drag the balloon with the W,A,S,D keys. The confusing bit is that the I can also get to the grab button and the black highlight is there, too, but the W,A,S,D keys do not work here. This makes sense now, and makes me even more confident that we can make this work based on fix in #187 as @jessegreenberg suggested we will be able to hide the button once the balloon is grabbed.

__Read through with ChromeVox (CV) cursor keys (Control-Command and Arrow keys)__
- The Two-balloon switch is read out as "Two-balloon Experiment, checkbox, not checked". CV does not recognize switch role yet.
- Everything else was the the same.

__Performance__
- Excellent performance with CV on Mac OS Sierra.

**Comment by jessegreenberg** [comment]
@terracoda can this issue be closed?  Would you like to keep these reviews documented somewhere or create new github issues for any problems you have found in the above testing?

**Comment by terracoda** [comment]
@jessegreenberg good point. I'll save the link to this issue in my todo document, so I get back to it easily.

And you can close it, as we completed what was necessary.

**Comment by terracoda** [comment]
@jessegreenberg I saved the link and made a note that we can review the testing to examine if there are other issues we need to work on.

**Comment by jessegreenberg** [comment]
Thanks @terracoda! I will close this issue.  If you would like to address any additional issues, feel free to create new issues and assign them to me.
